doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+For example, you could use "ancestor" and "direct parent" as a clear differentiation.
+
+There is also the term "(the) Parent Key".
+In many cases, this is refers to a very specific Key in the given context that is a "common parent" to a certain set of Keys.
+If you are working within such a context, be careful about "a parent Key" vs. "the Parent Key".
+
+### 1.2. Namespaces and Root Keys
+
+We mentioned above that there are different Namespaces in Elektra.
+Now we will explain their meaning.
+
+To recap, Elektra knows these Namespaces:
+
+- Cascading
+- Meta
+- Spec
+- Proc
+- Dir
+- User
+- System
+- Default
+
+We mentioned above that there are Key Names with zero Key Name Parts, i.e. just Namespace.
+These are called _Root Keys_ (based on Unix's filesystem root, as well as the root of a tree).
+
+Lets explore them one by one:
+
+- The simplest Namespace is the **"Meta"**.
+  The Namespace "Meta" is used exclusively for Meta Keys, i.e. Keys that are attached to another Key as metadata.
+  The Key Hierarchy of the Namespace "Meta" is entirely separate from the Key Hierarchies in the other Namespaces.
+  Without external information, you cannot determine from a Key Name with Namespace "Meta", which Key this Meta Key is attached to.
+- Keys with Namespace **"Proc"** only exist in memory and are scoped to the current <b>Proc</b>ess.
+- Keys with Namespace **"Dir"** are scoped to a single filesystem <b>Dir</b>ectory.
+  They will (normally) be stored somewhere within this directory.
+  Currently, there is no way of knowing which filesystem directory a Key with Namespace "Dir" is scoped to.
+  Elektra only uses Keys scoped to the Current Working Directory.
+- Similarly, Keys with Namespace **"User"** are scoped to a single **User**.
+  They will (normally) be stored somewhere within the User's home directory.
+  Again, there is no way of knowing which user a Key is scoped to and Elektra only uses Keys scoped to the current user, i.e. the one that executed the application.
+- The Namespace **"System"** is what makes Elektra global.
+  Keys with Namespace "System" are the same for all users of the system, independent of context.
+  They are stored in system level directory.
+- Keys with Namespace "Default" are special.
+  While you could create them manually, you normally don't want to.
+  It is used for Keys with default values for Namespace Resolution (explained below).
+  Keys with this Namespace are also in-memory only.
+- Then there is **"Spec"**.
+  This Namespace, like "Meta", is separate from the rest.
+  We use it for Keys that are part of a specification used to describe other Keys.
+  The Metadata of every Key with Namespace "Spec" describes a specification for the Keys that have the same Key Name Part but a different Namespace (except "Meta").
+  So the Namespace "Spec" has a closer relation to the others than "Meta".
+- Now just the Namespace **"Cascading"** remains.
+  This Namespace is used for Namespace Resolution (see below).
```

Comment:
If you separate special namespaces from other namespaces, you can write something like this:

```suggestion
  This namespace indicates that the key can be from any namespace.
```

Reply:
The statement 
> The namespace "Cascading" indicates that the key can be from any (other) namespace

doesn't make sense. Even with the word "other", the key in question **is** of the namespace "Cascading". This is a namespace just like any other and can be used as such. The only difference is it's meaning. But every namespace has very precise meaning, so it doesn't make sense to differentiate on that basis either.

The statement 

> A cascading key names can refer to a key from any namespace. Such keys names always refer to a single key, but which namespace this key has depends on the context.

makes a little more sense. But it is still inaccurate. You can create a `Key` with a cascading key name and use it entirely separately from a `KeySet`. It would only be accurate, if the cascading key names could only be used for `ksLookupByName` and not for `keyNew` or `keySetName`. But that is of course not possible because of limitations in the implementation.